src/directives/file-upload-bytes.js

```diff
@@ -0,0 +1,42 @@
+fileUploadBytesDirective.$inject = ['$rootScope'];
+function fileUploadBytesDirective($rootScope) {
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    scope: true,
+    link: function(scope, element, attrs, ngModel) {
+
+      scope.ngModel = ngModel;
+      scope.key = ngModel.$$scope.getKey()[ngModel.$$scope.getKey().length-1];
+      scope.fileInput = element[0].querySelector('.file-upload-field');
+      scope.inputText = "No File Chosen";
+      scope.fileName = null
+
+      scope.selectedFile = function () {
+        let file = document.getElementById(scope.key).files[0];
+        if (file) {
+          scope.fileName = file.name;
+        } else {
+          scope.fileName = null;
+        }
+        scope.validateValue();
+        scope.$apply();
+      }
+
+      scope.validateValue = function () {
+        Object.keys(ngModel.$error).forEach(function(k) {
```

Comment:
Can you explain what this particular loop is all about?  It's looping over the model errors and setting the validity to `true` for each of them?  Is that right? And why is that desirable / necessary?

Reply:
Okay so it looks like maybe this is to reset the validity state so that you're not stuck in a failed state if the validation fails once.  If that's correct, should this be resetting the validity for every error, or just the one that this validation function handles: "tv4-302"?  I would think the latter.

Reply:
This was how the validation worked in `file-upload.js` which was to reset to a valid state before validating, but the loop could be changed to this instead 
```
        if (!document.getElementById(scope.key).files[0] && ngModel.$$scope.form.required && !ngModel.$modelValue) {
          ngModel.$setValidity('tv4-302', false);
        } else {
          ngModel.$setValidity('tv4-302', true);
        }
```

Reply:
What you've suggested looks good.  That seems more clear and less prone to unintended side-effects.